foundation/org.eclipse.persistence.core/src/main/java/org/eclipse/persistence/platform/database/SQLServerPlatform.java

```diff
@@ -813,11 +813,11 @@ public void printSQLSelectStatement(DatabaseCall call, ExpressionSQLPrinter prin
             return;
         }
         
-        // OFFSET + FETCH NEXT requires ORDER BY, so add an ordering if there are none
-        // this SQL will satisfy the query parser without actually changing the ordering of the rows
+        // OFFSET + FETCH NEXT requires ORDER BY
         List<Expression> orderBy = statement.getOrderByExpressions();
         if (orderBy.isEmpty()) {
-            orderBy.add(statement.getBuilder().literal("ROW_NUMBER() OVER (ORDER BY (SELECT null))"));
+            super.printSQLSelectStatement(call, printer, statement);
+            return;
         }
```

Comment:
Sorry I'm not specialist for MS SQL, but could You please provide some example and simple SQL query with `ROW_NUMBER() OVER`
As I understand previous change in #300 did transformation like
origin query `SELECT ... FROM table`
into 'SELECT ... FROM table ORDER BY ROW_NUMBER() OVER (ORDER BY (SELECT null)) OFFSET ? ROWS'
or `SELECT ... FROM table ORDER BY ROW_NUMBER() OVER (ORDER BY (SELECT null)) OFFSET ? ROWS FETCH NEXT ? ROWS ONLY`
in SQL Server 2012 and higher.
Your change will keep query without order `SELECT ... FROM table` untouched and
query like `SELECT ... FROM table ORDER BY column1` will be transformed
into `SELECT ... FROM table ORDER BY column1 ? OFFSET ? ROWS` or
`SELECT ... FROM table ORDER BY column1 ? OFFSET ? ROWS FETCH NEXT ? ROWS ONLY`
and fix #300 was workaround for https://github.com/microsoft/mssql-jdbc/issues/176
Is it this change safe for issue with JDBC `Statement#setMaxRows()` mentioned in #300?